Apps/Sandcastle/gallery/Cesium OSM Buildings Styling.html

```diff
@@ -0,0 +1,260 @@
+                  "color('#FF5733')",
+                ],
+                ["true", "color('#FFC300')"],
+              ],
+            },
+          });
+        }
+
+        function colorByLatitude() {
+          buildings.style = new Cesium.Cesium3DTileStyle({
+            defines: {
+              lagitudeRadians: "radians(${feature['cesium#latitude']})",
+            },
+            color: {
+              conditions: [
+                ["${lagitudeRadians} >= 0.83117", "color('purple')"],
+                ["${lagitudeRadians} >= 0.83112", "color('red')"],
+                ["${lagitudeRadians} >= 0.83107", "color('orange')"],
+                ["${lagitudeRadians} >= 0.83102", "color('yellow')"],
+                ["${lagitudeRadians} >= 0.83097", "color('lime')"],
+                ["${lagitudeRadians} >= 0.83092", "color('cyan')"],
+                ["true", "color('blue')"],
+              ],
+            },
+          });
+        }
+
+        function colorByMaterial() {
+          buildings.style = new Cesium.Cesium3DTileStyle({
+            defines: {
+              material: "${feature['building:material']}",
+            },
+            color: {
+              conditions: [
+                ["${material} === null", "color('white')"],
+                ["${material} === undefined", "color('white')"],
+                ["${material} === 'glass'", "color('skyblue', 0.5)"],
+                ["${material} === 'concrete'", "color('grey')"],
+                ["${material} === 'brick'", "color('indianred')"],
+                ["${material} === 'stone'", "color('lightslategrey')"],
+                ["${material} === 'metal'", "color('lightgrey')"],
+                ["${material} === 'steel'", "color('lightsteelblue')"],
+                ["true", "color('white')"],
+              ],
+            },
+          });
+        }
+
+        function colorByType() {
+          buildings.style = new Cesium.Cesium3DTileStyle({
+            color: {
+              conditions: [
+                [
+                  "${feature['building']} === 'apartments' || ${feature['building']} === 'residential'",
+                  "color('cyan', 0.9)",
+                ],
+                [
+                  "${feature['building']} === 'school' || ${feature['building']} === 'university'",
+                  "color('lawngreen', 0.9)",
+                ],
+                [
+                  "${feature['building']} === 'hospital'",
+                  "color('yellow', 0.9)",
+                ],
+                [
+                  "${feature['building']} === 'construction'",
+                  "color('orangered', 0.9)",
+                ],
+                [
+                  "${feature['building']} === 'office' || ${feature['building']} === 'commercial'",
+                  "color('blueviolet', 0.9)",
+                ],
+                [true, "color('white')"],
+              ],
+            },
+          });
+        }
```

Comment:
I think this would be better if the title was "Highlight all residential buildings", and it was just the one condition. It's a good example of how you often need to check multiple things (apartments and residential) to highlight a particular type.

Reply:
I wanted to clarify your comment here. Does this mean we should get rid of other types of buildings and have this function shows highlighting apartments and residential buildings?

Reply:
Yeah, either hide all buildings except apartments/residential, or just color those ones differently, whichever you think looks best.